### PR TITLE
misc fixes to oauth related api's on botservice swagger

### PR DESCRIPTION
**Pull request by swagatmishra2007**
added some fixes to the api swagger found while testing the api. 
It involves making some items not readonly, since they can be modified by a client. 
also fixed an url.

### CODE REVIEW

**Comment by swagatmishra2007** [comment]
Failures in CI dont seem to be due to my changes. Can someone help with this?
Error: Cannot find module 'tslib'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)

**Review by marstr — CHANGES_REQUESTED** [review]
~~If you're removing resource group name, this parameter declaration should also be removed.~~
edit: embarrassingly commented in the wrong spot.

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json [hunk 1]
```diff
@@ -683,7 +683,7 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/BotServices/listServiceProviders": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.BotService/listAuthServiceProviders": {
```
Comment (marstr) [inline specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json]: Please remove the other reference to the resource group parameter in this operation:
https://github.com/Azure/azure-rest-api-specs/pull/3280/files#diff-6e7e52c7a9eb6243152b8870ce21ddf4R700

**Review by marstr — CHANGES_REQUESTED** [review]

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json (hunk 1 above)
Comment (marstr) [reply specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json]: This is the error that is currently causing CI to fail.

**Comment by marstr** [comment]
Just a friendly reminder that this PR is waiting for your attention, @swagatmishra2007. :)

File: specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json (hunk 1 above)
Comment (swagatmishra2007) [reply specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json]: i see...thanks for pointing it out. the error in the ci build seemed puzzling.

**Review by marstr — APPROVED** [review]

**Comment by marstr** [comment]
This is a de minimis is enough change that I do not believe it warrants ARM approval. Are you ready for me to merge it, @swagatmishra2007? i.e. are these changes ready for public preview consumption?

**Comment by swagatmishra2007** [comment]
@marstr  yes, these changes are ready for preview. please merge (and thanks for the review!)

**Pull request merged**